core/storage/statistics/gae_models.py

```diff
@@ -740,6 +969,21 @@ class StateAnswersCalcOutputModel(base_models.BaseMapReduceBatchResultsModel):
     @classmethod
     def create_or_update(cls, exploration_id, exploration_version, state_name,
                          calculation_id, calculation_output):
+        """Creates or updates StateAnswersCalcOutputModel and then writes
+        it on the datastore.
+
+        Args:
+            exploration_id: str. ID of the exploration currently being played.
+            exploration_version: int. Version of exploration.
+            state_name: str. Name of current state.
+            calculation_id: str. It tells about the type of calculation.
+            calculation_output: dict. Output of the calculation which is to be
+            stored as a Json blob.
+
+        Raises:
+            exception if the calculation_output is too large and it is unable
```

Comment:
The format for exceptions is a bit different. Check out for example line 357 of editor.py

Reply:
edited